pyproject.toml

```diff
@@ -268,16 +292,83 @@ optional = true
 [tool.poetry.dependencies.sentencepiece]
 version = "~0.1.96"
 optional = true
-extras = [ "sentencepiece",]
+extras = ["sentencepiece"]
 
 [tool.poetry.dependencies.jieba]
 version = ">=0.39, <0.43"
 optional = true
 
 [tool.poetry.dependencies.pymongo]
 version = ">=3.8,<3.11"
-extras = [ "tls", "srv",]
+extras = ["tls", "srv"]
 
 [tool.poetry.dev-dependencies.pytest-sanic]
 git = "https://github.com/RasaHQ/pytest-sanic"
 branch = "fix_signal_issue"
+
+[tool.mypy]
+# some libraries do not have type hints; ideally we remove
+# this at some point but it will require a lot of efforts from us and library owners
+mypy_path = "./stubs"
+ignore_missing_imports = true
+show_error_codes = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+disallow_untyped_calls = true
+disallow_incomplete_defs = true
+disallow_untyped_defs = true
+disallow_untyped_decorators = true
+# FIXME: should be switch to true (default) after fixing implicit Optional in the codebase
+no_implicit_optional = false
+# FIXME: working our way towards removing these
+# see https://github.com/RasaHQ/rasa/pull/6470
+# the list below is sorted by the number of errors for each error code, in decreasing order
+disable_error_code = ["arg-type"]
+
+[tool.pytest.ini_options]
+# Function starting with the following pattern are considered for test cases.
+python_functions = "test_"
+filterwarnings = ["ignore::ResourceWarning:ruamel[.*]"]
+log_cli_level = "WARNING"
+log_cli = true
+asyncio_mode = "auto"
```

Comment:
this option is not properly picked up, that's why tests are failing

Reply:
but it was already in this toml file, this is weird, I could be wrong

Reply:
might this be related to the async testing that thomas mentioned on SDK in the engine channel?

Reply:
I don't think it's the same problem. The problem with the SDK is that it was missing the `asyncio_mode = "auto"` (by default pytest-asyncio requires you to mark tests with `@pytest.marker.asyncio`). It was already present in the `toml` file (see above), but when I moved all the pytest config inside the `toml` file, it started to behave weirdly (it seems). I want on a goose chase to look at how pytest and plugins load their config, and I don't think it comes from there. So right now I don't know.

I've just merged `main` to see if Danil improvements change the game, but I don't have much hope. I'll try to move the config back to `pytest.ini` to see if it goes green again.

Reply:
😱 😱 😱 😱 you were totally right.... and we have the same problem on `rasa`. The config was scattered between `pyproject.toml`, `pytest.ini` and `setup.cfg`. So the `asyncio_mode = "auto"` directive had no effect because it wasn't picked up... what a mess. I can't fix it in this PR but I'll make sure to open a separate ticket....
<img width="919" alt="Screenshot 2023-03-17 at 15 31 37" src="https://user-images.githubusercontent.com/1931615/225944655-d2303522-a00e-45aa-ae82-7f4a303e12f6.png">